Mathlib/Algebra/Order/Ring/Finset.lean

```diff
@@ -1,17 +1,21 @@
 /-
-Copyright (c) 2022 Eric Wieser. All rights reserved.
+Copyright (c) 2022 Eric Wieser, Yaël Dillies, Andrew Yang. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Eric Wieser
+Authors: Eric Wieser, Yaël Dillies, Andrew Yang
 -/
-import Mathlib.Algebra.Order.Ring.Defs
+import Mathlib.Algebra.Order.Field.Canonical.Defs
+import Mathlib.Algebra.Order.Floor
 import Mathlib.Data.Finset.Lattice.Fold
 
 /-!
 # Algebraic properties of finitary supremum
 -/
 
 namespace Finset
-variable {ι R : Type*} [LinearOrderedSemiring R] {a b : ι → R}
+variable {ι R : Type*}
+
+section LinearOrderedSemiring
+variable [LinearOrderedSemiring R] {s : Finset ι} {a b : ι → R}
 
 theorem sup_mul_le_mul_sup_of_nonneg [OrderBot R] (s : Finset ι) (ha : ∀ i ∈ s, 0 ≤ a i)
     (hb : ∀ i ∈ s, 0 ≤ b i) : s.sup (a * b) ≤ s.sup a * s.sup b :=
```

Comment:
Don't these lemmas also work for something much weaker than semiring? It looks like this one needs less than `sup'_mul₀`.